### PR TITLE
Add example notebooks

### PR DESCRIPTION
**Pull request by gilad-shaham**

### CODE REVIEW

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: examples/v3io-kv.ipynb [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+    "> **Note**: You can pass to the client the `endpoint` and `access_key` parameters explicitly.\n",
+    "> The following code is equivalent to the default values:\n",
+    ">\n",
+    "> ``` python\n",
+    "> from os import getenv\n",
+    "> v3io_client = v3io.dataplane.Client(endpoint='http://v3io-webapi:8081',\n",
+    ">                                     access_key=getenv('V3IO_ACCESS_KEY'))\n",
+    "> ```\n",
+    ">\n",
+    "> When running the code remotely, you can obtain the URL of your cluster by copying the API URL of the web-APIs service (`webapi`) from the **Services** dashboard page. You can select between two types of URLs:\n",
```
Comment (assaf758) [inline examples/v3io-kv.ipynb]: missing information when running remotely:
- how to obtain the V3IO_ACCESS_KEY when running remotely
- how to obtain V3IO_USERNAME (used later)

File: examples/v3io-kv.ipynb [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime\n",
+    "from array import array\n",
```
Comment (assaf758) [inline examples/v3io-kv.ipynb]: no need to import the array package

File: examples/v3io-kv.ipynb [hunk 3]
```diff
@@ -0,0 +1,674 @@
...
+   "metadata": {},
+   "source": [
+    "## Put Multiple Items"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To get the highest possible throughput, you can send many requests towards the data layer and wait for all the responses to arrive (rather than send each request and wait for the response).\n",
```
Comment (assaf758) [inline examples/v3io-kv.ipynb]: I would rephrase to suggest to approaches for maximizing performance - batching and async (and point to the readme for async)

File: examples/v3io-kv.ipynb [hunk 4]
```diff
@@ -0,0 +1,674 @@
...
+   "metadata": {},
+   "source": [
+    "## Create a Schema (Optional)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To support reading and writing NoSQL data using structured-data interfaces &mdash; such as Spark DataFrames and Presto, the platform uses a schema file that defines the schema of the data structure.\n",
```
Comment (assaf758) [inline examples/v3io-kv.ipynb]: Presto => Trino

File: examples/v3io-kv.ipynb (hunk 1 above)
Comment (assaf758) [inline examples/v3io-kv.ipynb]: This is ambiguous  - the iguazio platform is "remote" from the perspective of the user.
I would say "when running python code on local machine that connects to remote iguazio platform..."

File: examples/v3io-kv.ipynb [hunk 5]
```diff
@@ -0,0 +1,674 @@
...
+       " (202, '1974-12-18 00:00:00.000', 9.0, 'The Godfather: Part II')]"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%sql select * from $query_table"
```
Comment (assaf758) [inline examples/v3io-kv.ipynb:617]: * the %sql magic command and the fuse commands used for delete are not available for when connecting to remote iguazio platform.
* You can show how to do the delete with list-dir (get-container-content) and item delete - don't know if you want to.
* Is connecting to remote iguazio a common use-case? If not I would consider not mentioning remote access, considering these limitations 
* ssh into jupyter is another matter of course - but it's probably out of scope here

File: examples/v3io-objects.ipynb [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+    ">\n",
+    "> ``` python\n",
+    "> from os import getenv\n",
+    "> v3io_client = v3io.dataplane.Client(endpoint='http://v3io-webapi:8081',\n",
+    ">                                     access_key=getenv('V3IO_ACCESS_KEY'))\n",
+    "> ```\n",
+    ">\n",
+    "> When running the code remotely, you can obtain the URL of your cluster by copying the API URL of the web-APIs service (`webapi`) from the **Services** dashboard page. You can select between two types of URLs:\n",
+    ">\n",
+    "> - **HTTPS Direct** (recommended) &mdash; a URL of the format `https://<tenant IP>:<web-APIs port>`; for example, `https://default-tenant.app.mycluster.iguazio.com:8443`.\n",
```
Comment (assaf758) [inline examples/v3io-objects.ipynb:65]: same comments on remote as in v3io-kv notebook

File: examples/v3io-objects.ipynb [hunk 7]
```diff
@@ -0,0 +1,543 @@
...
+   "outputs": [],
+   "source": [
+    "CONTAINER = 'users'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the data path for storing the NoSQL (KV) table:"
```
Comment (assaf758) [inline examples/v3io-objects.ipynb]: KV table => object store (or similar)

File: examples/v3io-objects.ipynb [hunk 8]
```diff
@@ -0,0 +1,543 @@
...
+   "source": [
+    "response = v3io_client.object.get(container=CONTAINER, path=path.join(OBJECTS_PATH, 'obj_00'))\n",
+    "print(response.body.decode('utf-8'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use the file system to list the new objects:"
```
Comment (assaf758) [inline examples/v3io-objects.ipynb]: again will not work when remote connecting

File: examples/v3io-streams.ipynb [hunk 9]
```diff
@@ -0,0 +1,901 @@
...
+    "                                   seek_type='EARLIEST')\n",
+    "location = seek_response.output.location"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use the `get_records` method to read from the stream (retrieve records).\n",
+    "By default `get_records` limits the number of records returned to 1,000.\n",
```
Comment (assaf758) [inline examples/v3io-streams.ipynb:328]: maybe mention that:
* get_records get the records from specific shard
* the seek and get() should be for the same shard.
* maybe point to the section on shard-id for those with sharp eye noticing entries are missing.

File: examples/v3io-streams.ipynb [hunk 10]
```diff
@@ -0,0 +1,901 @@
...
+   "source": [
+    "Use the `update` method to updates a stream's configuration by increasing its shard count.\n",
+    "The changes are applied immediately.\n",
+    "\n",
+    "> **Note**: You can increase the shard count at any time, but you cannot reduce it.\n",
+    "> From the platform's perspective, there's virtually no cost to creating even thousands of shards.\n",
+    "> However, if you increase a stream's shard count after its creation, new records with a previously used partition key will be assigned either to the same shard that was previously used for this partition key or to a new shard.\n",
+    "> For more information see the platform's [stream sharding and partitioning](https://www.iguazio.com/docs/latest-release/data-layer/stream/#stream-sharding-and-partitioning) documentation.\n",
+    "\n",
+    "> **Spark-Streaming Note**: To use the Spark Streaming API to consume records from new shards after a shard-count increase, you must first restart the consumer application."
```
Comment (assaf758) [inline examples/v3io-streams.ipynb]: how spark-streaming is related - I guess you mean v3io-stream, right?
I assume this comment is referencing nuclio consumer-groups - if you still want it here be more verbose

File: examples/v3io-streams.ipynb [hunk 11]
```diff
@@ -0,0 +1,901 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import shutil\n",
+    "V3IO_STREAM_PATH = path.join(path.sep, 'v3io', CONTAINER, STREAM_PATH)\n",
+    "shutil.rmtree(V3IO_STREAM_PATH)"
```
Comment (assaf758) [inline examples/v3io-streams.ipynb:872]: again does not work remote connecting.

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: examples/v3io-kv.ipynb (hunk 5 above)
Comment (gilad-shaham) [reply examples/v3io-kv.ipynb:617]: This is just to show that the data is readable with the schema that was created. Not sure if you have a better way, because this should be either spark or trino.
We could skip that "proof", but it does make the example more complete.

File: examples/v3io-streams.ipynb (hunk 10 above)
Comment (gilad-shaham) [reply examples/v3io-streams.ipynb]: Right, not sure why it says "spark"

File: examples/v3io-streams.ipynb (hunk 11 above)
Comment (gilad-shaham) [reply examples/v3io-streams.ipynb:872]: Given no API to delete, just added a note

File: examples/v3io-objects.ipynb (hunk 8 above)
Comment (gilad-shaham) [reply examples/v3io-objects.ipynb]: changed to container.list

File: examples/v3io-streams.ipynb (hunk 11 above)
Comment (assaf758) [reply examples/v3io-streams.ipynb:872]: actually for stream there is - and the notebook describes it (just above)

**Review by assaf758 — APPROVED** [review]
